addons/utm/models/utm_medium.py

```diff
@@ -26,11 +35,27 @@ def create(self, vals_list):
         return super().create(vals_list)
 
     @api.ondelete(at_uninstall=False)
-    def _unlink_except_utm_medium_email(self):
-        utm_medium_email = self.env.ref('utm.utm_medium_email', raise_if_not_found=False)
-        if utm_medium_email and utm_medium_email in self:
-            raise UserError(_(
-                "The UTM medium '%s' cannot be deleted as it is used in some main "
-                "functional flows, such as the recruitment and the mass mailing.",
-                utm_medium_email.name
-            ))
+    def _unlink_except_utm_medium_record(self):
+        for medium in self.REQUIRED_UTM_MEDIUMS_REF:
+            utm_medium = self.env.ref(medium, raise_if_not_found=False)
+            if utm_medium and utm_medium in self:
+                raise UserError(_(
+                    "Oops, you can't delete the Medium '%s'.\n"
+                    "Doing so would be like tearing down a load-bearing wall \u2014 not the best idea.",
+                    utm_medium.name
+                ))
+
+    def _fetch_or_create_utm_medium(self, name):
+        try:
+            return self.env.ref(f'utm.utm_medium_{name}')
+        except ValueError:
+            utm_medium = self.sudo().env['utm.medium'].create({
+                'name': self.REQUIRED_UTM_MEDIUMS_REF.get(f'utm.utm_medium_{name}', '')
```

Comment:
Maybe just fallback on name if people want to use this to create new "master" data ?
```suggestion
                'name': self.REQUIRED_UTM_MEDIUMS_REF.get(f'utm.utm_medium_{name}', name)
```